atsign-foundation/at_server
# fix: Add authorization check for enrollment in lookup verb handler

**- What I did**
- When user is authenticated vai the APKAM mechanism, when executing lookup verb, verify if the user has authorization to the key's namespace.
- Refactor lookup_verb_handler.dart to increase the readability.
- Update the tests in scan_verb_test.dart. Populated enrollmentId to the inbound connection to mimic the connection is authenticated via the APKAM and authorization check for namespaces enrolled is performed.

**- How I did it**
- When user authenticates via the APKAM mechanism, fetch enrollmentId from the connection metadata. Retrieve the enrolled namespaces and verify if user has permission for the key's namespace. If yes, return the data; else throw UnAuthorizedException.
- Refactor 'Lookup_verb_handler.dart' 

**- How to verify it**
- Add unit tests in lookup_verb_test.dart to assert the APKAM enrollment changes. Ran all unit tests and functional tests in local and tests pass.
- Add following scenario in scan verb unit test - A test to verify scan returns all keys when enrollment has *:rw access

**- Description for the changelog**
